src/watchdog.js

```diff
@@ -98,253 +96,115 @@ export default class Watchdog {
 			}
 		};
 
-		/**
-		 * Throttled save method. The `save()` method is called the specified `saveInterval` after `throttledSave()` is called,
-		 * unless a new action happens in the meantime.
-		 *
-		 * @private
-		 * @type {Function}
-		 */
-		this._throttledSave = throttle(
-			this._save.bind( this ),
-			typeof config.saveInterval === 'number' ? config.saveInterval : 5000
-		);
-
-		/**
-		 * The current editor instance.
-		 *
-		 * @private
-		 * @type {module:core/editor/editor~Editor}
-		 */
-		this._editor = null;
+		if ( !this._restart ) {
+			throw new Error(
+				'The Watchdog class was split into the abstract `Watchdog` class and the `EditorWatchdog` class. ' +
+				'Please, use `EditorWatchdog` if you have used the `Watchdog` class previously.'
+			);
+		}
 
 		/**
-		 * The editor creation method.
+		 * The creation method.
 		 *
-		 * @private
+		 * @protected
 		 * @member {Function} #_creator
 		 * @see #setCreator
 		 */
 
 		/**
-		 * The editor destruction method.
+		 * The destruction method.
 		 *
-		 * @private
+		 * @protected
 		 * @member {Function} #_destructor
 		 * @see #setDestructor
 		 */
-		this._destructor = editor => editor.destroy();
 
 		/**
-		 * The latest saved editor data represented as a root name -> root data object.
+		 * The handled instance.
 		 *
-		 * @private
-		 * @member {Object.<String,String>} #_data
-		 */
-
-		/**
-		 * The last document version.
-		 *
-		 * @private
-		 * @member {Number} #_lastDocumentVersion
+		 * @abstract
+		 * @protected
+		 * @member {Object|undefined} #_instance
 		 */
 
 		/**
-		 * The editor source element or data.
+		 * The method responsible for the instance restarting.
 		 *
-		 * @private
-		 * @member {HTMLElement|String|Object.<String|String>} #_elementOrData
+		 * @abstract
+		 * @protected
+		 * @method #_restart
 		 */
 
 		/**
-		 * The editor configuration.
+		 * Traverses the error context and the handled instance to find out whether the error should
+		 * be handled by the given instance.
 		 *
-		 * @private
-		 * @member {Object|undefined} #_config
+		 * @abstract
+		 * @protected
+		 * @method #_isErrorComingFromThisInstance
+		 * @param {module:utils/ckeditorerror~CKEditorError} error
 		 */
 	}
 
 	/**
-	 * The current editor instance.
-	 *
-	 * @readonly
-	 * @type {module:core/editor/editor~Editor}
-	 */
-	get editor() {
-		return this._editor;
-	}
-
-	/**
-	 * Sets the function that is responsible for the editor creation.
-	 * It expects a function that should return a promise.
+	 * Sets the function that is responsible for the instance creation.
 	 *
-	 *		watchdog.setCreator( ( element, config ) => ClassicEditor.create( element, config ) );
-	 *
-	 * @param {Function} creator
+	 * @param {Function} creator A callback returning promise that is responsible for instance creation.
```

Comment:
Now it sounds that the promise is responsible for the instance creation. Please reword it -> "A callback responsible for creating instance. Returns a Promise that is resolved when the instance is created".